packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/SentenceScopeHandler.ts

```diff
@@ -0,0 +1,49 @@
+import { Direction, Range } from "@cursorless/common";
+import { imap } from "itertools";
+import { NestedScopeHandler } from ".";
+import type { TargetScope } from "./scope.types";
+import { TokenTarget } from "../../targets";
+
+export default class SentenceScopeHandler extends NestedScopeHandler {
+  public readonly scopeType = { type: "sentence" } as const;
+  public readonly iterationScopeType = { type: "paragraph" } as const;
+
+  protected generateScopesInSearchScope(
+    direction: Direction,
+    { editor, domain }: TargetScope,
+  ): Iterable<TargetScope> {
+    const segmenter = new Intl.Segmenter("en", { granularity: "sentence" });
+    const offset = editor.document.offsetAt(domain.start);
+    const text = editor.document.getText(domain);
+
+    const segmentToRange = (segment: Intl.SegmentData) => {
+      const leadingWhitespace = segment.segment.match(/^\s+/) ?? "";
+      const trailingWhitespace = segment.segment.match(/\s+$/) ?? "";
+      const start = offset + segment.index + leadingWhitespace.length;
+      return new Range(
+        editor.document.positionAt(start),
+        editor.document.positionAt(
+          start +
+            segment.segment.length -
+            leadingWhitespace.length -
+            trailingWhitespace.length,
+        ),
+      );
```

Comment:
maybe we should just have a `trimRange` function?  I think that would make it clearer what's happening here.  Plus we could use it for a `#trim!` query predicate in the future 😊

Reply:
I kinda had the same thought. I can do that.

Reply:
done